lib/semian/atomic_integer.rb

```diff
@@ -0,0 +1,28 @@
+module Semian
+  class AtomicInteger #:nodoc:
+    attr_accessor :value
+
+    def initialize(_name, _permissions)
+      @value = 0
+    end
+
+    def increment_by(val)
+      self.value += val
+    end
+
+    def increment
+      increment_by(1)
+    end
+
+    def execute_atomically
+      yield if block_given?
+    end
+
+    def shared?
+      false
+    end
```

Comment:
Yes, essentially I want something that lets me differentiate between a structure that has shared memory (and therefore needs a different set of methods for `destroy` (since we now need to free the semaphore, etc), `execute_atomically` (which locks, calls yield, and unlocks), but because we now have a fallback implementation as well, this functionality can't just be a complete override, so we need some `bool` of some sort to determine whether we are actually using shared memory. Thats what the `#shared?` was for. 

I can concede the iterating over all the classes for testing part, but that `bool` has to be there for things like `destroy` and `execute_atomically`. `#shared?` can be private if that seems like a better idea.